### ISSUE TITLE
See-through gaps between cells background on some systems

### ISSUE THREAD

**Issue by MightyPork** [espterm/espterm-firmware#171]
partly mitigated by setting background color to the most common cell bg, but it's still visible in some cases, like this:

![screenshot_20170913_225123](https://user-images.githubusercontent.com/2041118/30400547-88b9f398-98d7-11e7-9367-c549d2b8ac89.png)

if this gets fixed we can also remove the background color counting which certainly ain't making the load routine any faster

**Comment by c-probably-archived20260718** [comment]
The bug is caused due to the devicePixelRatio being an ugly value like 1.7999999523162842 (at e.g. 90% zoom) and because I set that as the transform and do nothing else, it's technically the browser's fault. Ways to fix it:

- ~round the DPR to a nice value (e.g. 1.8)~ ✘ still happens
- ~handle scaling manually~ ✘ still happens (or messes with redrawing)
- round the DPR to a very nice value (e.g. 1, 1.5, 2) and scale the canvas to fit ✓

Needs testing though

**Comment by MightyPork** [comment]
Guenter reported it happens also without zoom, so that's not quite adding
up.

If you make some test page to do measurements (and like, put the results in
a textarea to copy), I'll send him a link

**Comment by c-probably-archived20260718** [comment]
some screens have a 1.5 or 1.75 DPR by default so that might be it. And what would I measure anyway? The issue is that the browser's ctx.setTransform doesn't really work well for weird floaty values, and since I set the transform to the DPR, that should be the only relevant variable here. Though since I forgot to remove code I used for debugging, red lines will seep through the gaps if the bug should still happen (removed in espterm/espterm-front-end@a9f5948)
